parl/agent_zoo/agents.py

```diff
@@ -0,0 +1,74 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import numpy as np
+from parl.common.logging import GameLogEntry
+from parl.framework.agent import AgentBase
+
+
+class SimpleRLAgent(AgentBase):
+    """
+    This class serves as a template of simple RL algorithms, which has only one
+    ComputationTask, "RL", i.e., using and learning an RL policy. 
+    
+    By using different AgentHelpers, this Agent can be applied to either on-
+    policy or off-policy RL algorithms.
+    """
+
+    def __init__(self, env, num_games):
+        super(SimpleRLAgent, self).__init__(env, num_games)
+
+    def _run_one_episode(self):
+        # sensor_inputs, (prev_)states and actions are all dict
+        max_steps = self.env._max_episode_steps
+        obs = self.env.reset()
+        episode_end = False
+        r = 0
+        log_entry = GameLogEntry(self.id, 'RL')
+        for t in range(max_steps - 1):
+            #self.env.render()
+            sensor = np.array(obs).astype("float32")
+            inputs = dict(sensor=sensor)
+            actions, _ = self.helpers['RL'].predict(inputs)
```

Comment:
We should hide the concept of agent_helper behind agent. The user should call agent.predict("RL", inputs) instead of having to know the existence of self.helpers.